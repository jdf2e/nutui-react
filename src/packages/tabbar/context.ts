import React from 'react'

export interface TabbarContext {
  selectIndex: number
  inactiveColor: string
  activeColor: string
  handleClick: (value: number) => void
}

export default React.createContext<TabbarContext | null>(null)
