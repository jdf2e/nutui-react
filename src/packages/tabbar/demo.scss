#app {
  .demo {
    .bottom-h2 {
      margin-bottom: 60px;
    }
  }
}
