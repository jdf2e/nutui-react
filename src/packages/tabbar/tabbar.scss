@import '../tabbaritem/tabbaritem.scss';

.nut-tabbar {
  border: 0px;
  box-shadow: $tabbar-box-shadow;
  border-bottom: $tabbar-border-bottom;
  border-top: $tabbar-border-top;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  background: $white;

  &__wrap {
    width: 100%;
    height: $tabbar-height;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }

  &:last-child {
    border-right: 0;
  }

  &__fixed {
    position: fixed;
    bottom: 0px;
    left: 0px;
  }

  &__safe-area {
    display: block;
    width: 100%;
    padding-bottom: constant(safe-area-inset-bottom);
    padding-bottom: env(safe-area-inset-bottom);
  }
}
