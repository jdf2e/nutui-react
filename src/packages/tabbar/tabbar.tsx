import React, { FunctionComponent, ReactComponentElement, useState } from 'react'

import bem from '@/utils/bem'

export interface TabbarProps {
  visible: number | string
  bottom: boolean
  type: string
  size: string
  unactiveColor: string
  activeColor: string
  className: string
  style: React.CSSProperties
  tabSwitch: (child: React.ReactElement<any>, active: number) => void
}
const defaultProps = {
  visible: 0,
  bottom: false,
  type: '',
  size: '',
  unactiveColor: '',
  activeColor: '',
  className: '',
  style: {},
  tabSwitch: () => {},
} as TabbarProps

export const Tabbar: FunctionComponent<Partial<TabbarProps>> = (props) => {
  const { children, visible, bottom, activeColor, unactiveColor, className, style, tabSwitch } = {
    ...defaultProps,
    ...props,
  }

  const b = bem('tabbar')

  const [selectIndex, setSelectIndex] = useState(visible)

  const handleClick = (idx: number) => {
    setSelectIndex(idx)
  }

  return (
    <div className={[`${b()}`, bottom ? `${b('bottom')}` : '', className].join(' ')} style={style}>
      {React.Children.map(children, (child, idx) => {
        if (!React.isValidElement(child)) {
          return null
        }
        const childProps = {
          ...child.props,
          active: idx === selectIndex,
          index: idx,
          unactiveColor: unactiveColor,
          activeColor: activeColor,
          handleClick: () => {
            handleClick(idx)
            tabSwitch(child, idx)
          },
        }
        return React.cloneElement(child, childProps)
      })}
    </div>
  )
}
Tabbar.defaultProps = defaultProps
Tabbar.displayName = 'NutTabbar'
