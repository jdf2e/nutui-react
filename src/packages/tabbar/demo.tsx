import React, { useState } from 'react'
import { Cart, Category, Find, Home, User } from '@nutui/icons-react'
import { Tabbar } from './tabbar'
import { useTranslate } from '../../sites/assets/locale'

interface T {
  ce5c5446: string
  c38a08ef: string
  ce5c5448: string
  ce5c5440: string
  b840c88f: string
  a74a1fd4: string
  '8dab2f66': string
  cfbdc781: string
  c3a3a1d2: string
  e51e4582: string
  '7db1a8b2': string
  a52bef0c: string
  d04fcbda: string
  c9e6df49: string
}

const TabbarDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      ce5c5446: '基础用法',
      c38a08ef: '自定义选中',
      ce5c5448: '只配图标',
      ce5c5440: '无图标',
      b840c88f: '徽标提示',
      a74a1fd4: '自定义颜色',
      '8dab2f66': '可自定义icon个数的tabbar',
      cfbdc781: '固定底部',
      c3a3a1d2: '首页',
      e51e4582: '我的',
      '7db1a8b2': '购物车',
      a52bef0c: '发现',
      d04fcbda: '分类',
      c9e6df49: '红点',
    },
    'zh-TW': {
      ce5c5446: '基礎用法',
      c38a08ef: '自定義選中',
      ce5c5448: '只配图标',
      ce5c5440: '无图标',
      b840c88f: '徽標提示',
      a74a1fd4: '自定義顏色',
      '8dab2f66': '可自定義icon個數的tabbar',
      cfbdc781: '固定底部',
      c3a3a1d2: '首頁',
      e51e4582: '我的',
      '7db1a8b2': '購物車',
      a52bef0c: '發現',
      d04fcbda: '分類',
      c9e6df49: '紅點',
    },
    'en-US': {
      ce5c5446: 'Basic Usage',
      c38a08ef: 'Custom DefaultValue',
      ce5c5448: 'Only Icon',
      ce5c5440: 'No Icon',
      b840c88f: 'Logo Tips',
      a74a1fd4: 'Custom Color',
      '8dab2f66': 'Tabbar with customizable number of icons',
      cfbdc781: 'Fixed bottom',
      c3a3a1d2: 'Home',
      e51e4582: 'Mine',
      '7db1a8b2': 'Cart',
      a52bef0c: 'Find',
      d04fcbda: 'Category',
      c9e6df49: 'Dot',
    },
  })

  const [activeIndex, setActiveIndex] = useState(2)

  return (
    <>
      <div className="demo">
        <h2>{translated.ce5c5446}</h2>
        <Tabbar defaultValue={0}>
          <Tabbar.Item
            title={translated.c3a3a1d2}
            icon={<Home width={18} height={18} />}
            value={9}
          />
          <Tabbar.Item
            title={translated.d04fcbda}
            icon={<Category width={18} height={18} />}
            dot
          />
          <Tabbar.Item
            title={translated.a52bef0c}
            icon={<Find width={18} height={18} />}
          />
          <Tabbar.Item
            title={translated['7db1a8b2']}
            icon={<Cart width={18} height={18} />}
          />
          <Tabbar.Item
            title={translated.e51e4582}
            icon={<User width={18} height={18} />}
          />
        </Tabbar>

        <h2>{translated.c38a08ef}</h2>
        <Tabbar
          defaultValue={0}
          value={activeIndex}
          onSwitch={(value) => {
            setActiveIndex(value)
          }}
        >
          <Tabbar.Item
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.e51e4582}
            icon={<User width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated.ce5c5448}</h2>
        <Tabbar
          onSwitch={(value) => {
            console.log(value)
          }}
        >
          <Tabbar.Item icon={<Home width={24} height={24} />} />
          <Tabbar.Item icon={<Category width={24} height={24} />} />
          <Tabbar.Item icon={<Find width={24} height={24} />} />
          <Tabbar.Item icon={<Cart width={24} height={24} />} />
          <Tabbar.Item icon={<User width={24} height={24} />} />
        </Tabbar>

        <h2>{translated.ce5c5440}</h2>
        <Tabbar
          onSwitch={(value) => {
            console.log(value)
          }}
        >
          <Tabbar.Item title={translated.c3a3a1d2} value={9} />
          <Tabbar.Item title={translated.d04fcbda} dot />
          <Tabbar.Item title={translated.a52bef0c} />
          <Tabbar.Item title={translated['7db1a8b2']} />
          <Tabbar.Item title={translated.e51e4582} />
        </Tabbar>

        <h2>{translated.b840c88f}</h2>
        <Tabbar>
          <Tabbar.Item title={translated.c3a3a1d2} icon={<Home />} value={11} />
          <Tabbar.Item title={translated.d04fcbda} icon={<Category />} />
          <Tabbar.Item title={translated.a52bef0c} icon={<Find />} />
          <Tabbar.Item
            title={translated['7db1a8b2']}
            icon={<Cart />}
            value={110}
          />
          <Tabbar.Item
            title={translated.e51e4582}
            icon={<User width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated.c9e6df49}</h2>
        <Tabbar>
          <Tabbar.Item
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
            dot
          />
          <Tabbar.Item
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
            dot
          />
          <Tabbar.Item
            title={translated.e51e4582}
            icon={<User width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated.a74a1fd4}</h2>
        <Tabbar inactiveColor="#7d7e80" activeColor="#1989fa">
          <Tabbar.Item
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.e51e4582}
            icon={<User width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated['8dab2f66']}</h2>
        <Tabbar inactiveColor="#7d7e80" activeColor="#1989fa">
          <Tabbar.Item title={translated.c3a3a1d2} icon={<Home />} />
          <Tabbar.Item title={translated.d04fcbda} icon={<Category />} />
          <Tabbar.Item title={translated.a52bef0c} icon={<Find />} />
        </Tabbar>

        <h2 style={{ marginBottom: '100px' }}>{translated.cfbdc781}</h2>
        <Tabbar fixed>
          <Tabbar.Item
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
          />
          <Tabbar.Item
            title={translated.e51e4582}
            icon={<User width={20} height={20} />}
          />
        </Tabbar>
      </div>
    </>
  )
}

export default TabbarDemo
