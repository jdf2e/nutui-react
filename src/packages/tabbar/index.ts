import { Tabbar } from './tabbar'

export type { TabbarContext } from './context'
export type { TabbarProps } from './tabbar'
export default Tabbar
