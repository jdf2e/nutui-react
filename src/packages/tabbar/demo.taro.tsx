import React from 'react'
import Taro from '@tarojs/taro'
import { Cart, Category, Find, Home, My } from '@nutui/icons-react-taro'
import { TabbarItem, Tabbar } from '@/packages/nutui.react.taro'
import '@/packages/tabbar/demo.scss'
import { useTranslate } from '@/sites/assets/locale/taro'
import Header from '@/sites/components/header'

interface T {
  ce5c5446: string
  c38a08ef: string
  ce5c5448: string
  ce5c5440: string
  b840c88f: string
  a74a1fd4: string
  '8dab2f66': string
  cfbdc781: string
  c3a3a1d2: string
  e51e4582: string
  '7db1a8b2': string
  a52bef0c: string
  d04fcbda: string
  c9e6df49: string
}

const TabbarDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      ce5c5446: '基础用法',
      c38a08ef: '自定义选中',
      ce5c5448: '只配图标',
      ce5c5440: '无图标',
      b840c88f: '徽标提示',
      a74a1fd4: '自定义颜色',
      '8dab2f66': '可自定义icon个数的tabbar',
      cfbdc781: '固定底部，可自由跳转',
      c3a3a1d2: '首页',
      e51e4582: '我的',
      '7db1a8b2': '购物车',
      a52bef0c: '发现',
      d04fcbda: '分类',
      c9e6df49: '红点',
    },
    'zh-TW': {
      ce5c5446: '基礎用法',
      c38a08ef: '自定義選中',
      ce5c5448: '只配图标',
      ce5c5440: '无图标',
      b840c88f: '徽標提示',
      a74a1fd4: '自定義顏色',
      '8dab2f66': '可自定義icon個數的tabbar',
      cfbdc781: '固定底部，可自由跳轉',
      c3a3a1d2: '首頁',
      e51e4582: '我的',
      '7db1a8b2': '購物車',
      a52bef0c: '發現',
      d04fcbda: '分類',
      c9e6df49: '紅點',
    },
    'en-US': {
      ce5c5446: 'Basic usage',
      c38a08ef: 'custom check',
      ce5c5448: 'some only Icon',
      ce5c5440: 'no icon',
      b840c88f: 'Logo Tips',
      a74a1fd4: 'custom color',
      '8dab2f66': 'Tabbar with customizable number of icons',
      cfbdc781: 'Fixed bottom, free to jump',
      c3a3a1d2: 'first',
      e51e4582: 'mine',
      '7db1a8b2': 'cart',
      a52bef0c: 'Find',
      d04fcbda: 'Classification',
      c9e6df49: 'Red dot',
    },
  })
  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <h2>{translated.ce5c5446}</h2>
        <Tabbar
          onSwitch={(child, idx) => {
            console.log(idx)
          }}
        >
          <TabbarItem title={translated.c3a3a1d2} icon={<Home size={18} />} />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category size={18} />}
          />
          <TabbarItem title={translated.a52bef0c} icon={<Find size={18} />} />
          <TabbarItem
            title={translated['7db1a8b2']}
            icon={<Cart size={18} />}
          />
          <TabbarItem title={translated.e51e4582} icon={<My size={18} />} />
        </Tabbar>

        <h2>{translated.c38a08ef}</h2>
        <Tabbar visible={2}>
          <TabbarItem
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
          />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <TabbarItem
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <TabbarItem
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
          />
          <TabbarItem
            title={translated.e51e4582}
            icon={<My width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated.ce5c5448}</h2>
        <Tabbar
          onSwitch={(child, idx) => {
            console.log(idx)
          }}
        >
          <TabbarItem title={translated.c3a3a1d2} icon={<Home size={12} />} />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category size={12} />}
          />
          <TabbarItem icon={<Find size={24} />} />
          <TabbarItem
            title={translated['7db1a8b2']}
            icon={<Cart size={12} />}
          />
          <TabbarItem title={translated.e51e4582} icon={<My size={12} />} />
        </Tabbar>

        <h2>{translated.ce5c5440}</h2>
        <Tabbar
          onSwitch={(child, idx) => {
            console.log(idx)
          }}
        >
          <TabbarItem title={translated.c3a3a1d2} />
          <TabbarItem title={translated.d04fcbda} />
          <TabbarItem title={translated.a52bef0c} />
          <TabbarItem title={translated['7db1a8b2']} />
          <TabbarItem title={translated.e51e4582} />
        </Tabbar>

        <h2>{translated.b840c88f}</h2>
        <Tabbar>
          <TabbarItem
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
            num="11"
          />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <TabbarItem
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <TabbarItem
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
            num="110"
          />
          <TabbarItem
            title={translated.e51e4582}
            icon={<My width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated.c9e6df49}</h2>
        <Tabbar>
          <TabbarItem
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
            dot
          />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <TabbarItem
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <TabbarItem
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
            dot
          />
          <TabbarItem
            title={translated.e51e4582}
            icon={<My width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated.a74a1fd4}</h2>
        <Tabbar unactiveColor="#7d7e80" activeColor="#1989fa">
          <TabbarItem
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
          />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <TabbarItem
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <TabbarItem
            title={translated['7db1a8b2']}
            icon={<Cart width={20} height={20} />}
          />
          <TabbarItem
            title={translated.e51e4582}
            icon={<My width={20} height={20} />}
          />
        </Tabbar>

        <h2>{translated['8dab2f66']}</h2>
        <Tabbar unactiveColor="#7d7e80" activeColor="#1989fa">
          <TabbarItem
            title={translated.c3a3a1d2}
            icon={<Home width={20} height={20} />}
          />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <TabbarItem
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
        </Tabbar>

        <h2 className="bottom-h2">{translated.cfbdc781}</h2>
        <Tabbar safeAreaInsetBottom>
          <TabbarItem
            title={translated.c3a3a1d2}
            href=""
            icon={<Home width={20} height={20} />}
          />
          <TabbarItem
            title={translated.d04fcbda}
            icon={<Category width={20} height={20} />}
          />
          <TabbarItem
            title={translated.a52bef0c}
            icon={<Find width={20} height={20} />}
          />
          <TabbarItem
            title={translated['7db1a8b2']}
            // href="https://m.jd.com"
            icon={<Cart width={20} height={20} />}
          />
          <TabbarItem
            title={translated.e51e4582}
            to="/pages/index/index"
            icon={<My width={20} height={20} />}
          />
        </Tabbar>
      </div>
    </>
  )
}

export default TabbarDemo
