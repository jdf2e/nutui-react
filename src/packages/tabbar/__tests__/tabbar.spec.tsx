import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Cart, Category, Find, Home, My } from '@nutui/icons-react'
import { Tabbar } from '../tabbar'
import { TabbarItem } from '../../tabbaritem/tabbaritem'

test('should render tabbar when default', () => {
  const { container } = render(
    <>
      <Tabbar>
        <TabbarItem title="首页" icon={<Home width={20} height={20} />} />
        <TabbarItem title="分类" icon={<Category width={20} height={20} />} />
        <TabbarItem title="发现" icon={<Find width={20} height={20} />} />
        <TabbarItem title="购物车" icon={<Cart width={20} height={20} />} />
        <TabbarItem title="我的" icon={<My width={20} height={20} />} />
      </Tabbar>
    </>
  )

  expect(container.firstChild).toBeInTheDocument()
  expect(container.querySelectorAll('.nut-tabbar-item').length).toEqual(5)
  expect(
    container.querySelectorAll('.nut-tabbar-item__icon-box').length
  ).toEqual(10)
})

test('should render custom color and badge when using prop', () => {
  const { container } = render(
    <>
      <Tabbar unactiveColor="grey" activeColor="blue">
        <TabbarItem
          title="首页"
          icon={<Home width={20} height={20} />}
          num="11"
        />
        <TabbarItem title="分类" icon={<Category width={20} height={20} />} />
        <TabbarItem title="发现" icon={<Find width={20} height={20} />} />
      </Tabbar>
    </>
  )

  const tabbarItem: NodeListOf<HTMLElement> =
    container.querySelectorAll('.nut-tabbar-item')

  expect(tabbarItem[0].style.color).toEqual('blue')
  expect(tabbarItem[1].style.color).toEqual('grey')
  expect(
    tabbarItem[0].querySelectorAll('.nut-tabbar-item__icon-box__tips').length
  ).toBeGreaterThan(0)
})

test('should render fixed element when using bottom prop', async () => {
  const { container } = render(
    <>
      <Tabbar bottom safeAreaInsetBottom>
        <TabbarItem title="首页" icon={<Home width={20} height={20} />} />
        <TabbarItem title="分类" icon={<Category width={20} height={20} />} />
      </Tabbar>
    </>
  )
  expect(container.innerHTML).toMatchSnapshot()
})

test('should match active tabbar by click', async () => {
  const { container } = render(
    <>
      <Tabbar unactiveColor="grey" activeColor="blue">
        <TabbarItem
          title="首页"
          icon={<Home width={20} height={20} />}
          num="11"
        />
        <TabbarItem title="分类" icon={<Category width={20} height={20} />} />
        <TabbarItem title="发现" icon={<Find width={20} height={20} />} />
      </Tabbar>
    </>
  )

  const tabbarItem: NodeListOf<HTMLElement> =
    container.querySelectorAll('.nut-tabbar-item')
  expect(tabbarItem[0].style.color).toEqual('blue')
  fireEvent.click(tabbarItem[1])
  expect(tabbarItem[1].style.color).toEqual('blue')
  fireEvent.click(tabbarItem[2])
  expect(tabbarItem[2].style.color).toEqual('blue')
})

test('should show sure emitted when click', async () => {
  const onSwitch = jest.fn()
  const { container } = render(
    <>
      <Tabbar unactiveColor="grey" activeColor="blue" onSwitch={onSwitch}>
        <TabbarItem
          title="首页"
          icon={<Home width={20} height={20} />}
          num="11"
        />
        <TabbarItem title="分类" icon={<Category width={20} height={20} />} />
        <TabbarItem title="发现" icon={<Find width={20} height={20} />} />
      </Tabbar>
    </>
  )

  const tabbarItem: NodeListOf<HTMLElement> =
    container.querySelectorAll('.nut-tabbar-item')
  fireEvent.click(tabbarItem[1])
  expect(onSwitch).toBeCalled()
})
