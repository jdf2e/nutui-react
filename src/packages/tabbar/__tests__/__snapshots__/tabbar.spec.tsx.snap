// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should only render title 1`] = `"<div class="nut-tabbar"><div class="nut-tabbar-wrap"><div class="nut-tabbar-item nut-tabbar-item-active" style="color: blue;"><div class="nut-badge"><div class="nut-tabbar-item-icon-box nut-tabbar-item-icon-box-nav nut-tabbar-item-icon-box-large">首页</div><div class="nut-badge-content nut-badge-sup" style="top: 0px; right: 5px; background: blue;">11</div></div></div><div class="nut-tabbar-item" style="color: grey;"><div class="nut-badge"><div class="nut-tabbar-item-icon-box nut-tabbar-item-icon-box-nav nut-tabbar-item-icon-box-large">分类</div><div class="nut-badge-content" style="top: 0px; right: 5px; background: blue;"></div></div></div><div class="nut-tabbar-item" style="color: grey;"><div class="nut-badge"><div class="nut-tabbar-item-icon-box nut-tabbar-item-icon-box-nav nut-tabbar-item-icon-box-large">发现</div><div class="nut-badge-content" style="top: 0px; right: 5px; background: blue;"></div></div></div></div></div>"`;

exports[`should render fixed element when using bottom prop 1`] = `"<div class="nut-tabbar nut-tabbar-fixed"><div class="nut-tabbar-wrap"><div class="nut-tabbar-item nut-tabbar-item-active"><div class="nut-badge"><div class="nut-tabbar-item-icon-box"><svg class="nut-icon nut-icon-Home " style="width: 20px; height: 20px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Home" role="presentation"><path d="M761.44 1024H262.56c-131.28 0-236.31-105.03-236.3-236.31V404.35c0-52.51 26.26-102.4 68.26-131.28L467.36 21.01c26.26-18.38 60.39-18.38 89.28 0l372.84 252.06c42.01 28.88 68.27 78.77 68.26 131.28v383.34c0 131.28-105.03 236.31-236.3 236.31zM139.16 338.71c-21 15.75-34.13 39.38-34.13 65.64v383.34c0 86.65 70.89 157.54 157.53 157.54h498.88c86.65 0 157.54-70.89 157.53-157.54V404.35c0-26.26-13.13-49.89-34.13-65.64L512 86.65 139.16 338.71zm451.61 396.47H433.23c-21 0-39.38-18.38-39.38-39.38s18.38-39.38 39.38-39.39h157.54c21 0 39.38 18.38 39.38 39.39s-18.38 39.38-39.38 39.38z" fill="currentColor" fill-opacity="0.9"></path></svg></div><div class="nut-badge-content" style="top: 0px; right: 5px;"></div></div><div class="nut-tabbar-item-icon-box nut-tabbar-item-icon-box-nav">首页</div></div><div class="nut-tabbar-item"><div class="nut-badge"><div class="nut-tabbar-item-icon-box"><svg class="nut-icon nut-icon-Category " style="width: 20px; height: 20px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Category" role="presentation"><path d="M298.67 170.67a128 128 0 1 1-256 0 128 128 0 0 1 256 0zm-85.34 0a42.67 42.67 0 1 0-85.33 0 42.67 42.67 0 0 0 85.33 0zm85.34 682.66a128 128 0 1 1-256 0 128 128 0 0 1 256 0zm-85.34 0a42.67 42.67 0 1 0-85.33 0 42.67 42.67 0 0 0 85.33 0zM170.67 640a128 128 0 1 0 0-256 128 128 0 0 0 0 256zm0-85.33a42.67 42.67 0 1 1 0-85.34 42.67 42.67 0 0 1 0 85.34zm256-426.67a42.67 42.67 0 1 0 0 85.33h512a42.67 42.67 0 1 0 0-85.33h-512zM384 512a42.67 42.67 0 0 1 42.67-42.67h512a42.67 42.67 0 1 1 0 85.34h-512A42.67 42.67 0 0 1 384 512zm42.67 298.67a42.67 42.67 0 1 0 0 85.33h512a42.67 42.67 0 1 0 0-85.33h-512z" fill="currentColor" fill-opacity="0.9"></path></svg></div><div class="nut-badge-content" style="top: 0px; right: 5px;"></div></div><div class="nut-tabbar-item-icon-box nut-tabbar-item-icon-box-nav">分类</div></div></div><div class="nut-tabbar-safe-area"></div></div>"`;
